src/main/kotlin/net/rrerr/minetrix/PlayerEventListener.kt

```diff
@@ -37,4 +38,18 @@ class PlayerEventListener(plugin: Main) : ListenerBase(plugin) {
         val message = "☠ ${ChatColor.stripColor(ev.deathMessage!!)}"
         sendMessage(TextMessageContent(message))
     }
+
+    @EventHandler
+    fun playerAdvancementDone(ev: PlayerAdvancementDoneEvent) {
+        // We only want to forward advancements that are done, are not some new unlocked recipes,
+        // and are not the start of some advancement chain.
+        if (!ev.player.getAdvancementProgress(ev.advancement).isDone ||
+            ev.advancement.key.key.startsWith("recipes/") ||
+            ev.advancement.key.key.endsWith("/root")) {
+            return
+        }
+
+        val message = "🏆 ${ev.player.name} has made the advancement ${ev.advancement.key.key}"
```

Comment:
There exitsts a `TranslatableComponent` which can be used to look up the advancement name.

In fact you could use `chat.type.advancement.task` for the message text
https://github.com/SpigotMC/BungeeCord/blob/master/chat/src/main/resources/mojang-translations/en_US.properties#L301

Reply:
Although this would involve also changing the other messages.

Reply:
Hm, how would I be able to use `TranslatableComponent` for this?  IIRC these component classes only work with the Minecraft in-game chat, and not with anything else 🤔

Reply:
From looking around it would seem you can get the localised names from: `TranslationRegistry.INSTANCE.translate(key)`

https://github.com/SpigotMC/BungeeCord/blob/5aaccd2e9ec554f8f769aa0eb11a7393236aad85/chat/src/main/java/net/md_5/bungee/api/chat/TranslatableComponent.java#L142

Reply:
Will give this a try, thanks!

Reply:
For adding the original Minecraft text as the message which gets posted into Matrix, you could use that `toPlainText` method to generate the whole message. Although we could leave that for later.